src/physics/advection_driver.f90

```diff
@@ -63,17 +75,54 @@ subroutine advect(domain,options,dt)
         !     allocate(domain%tend%qv_adv(nx,nz,ny))
         !     domain%tend%qv_adv=0
         ! endif
+        
+        !Allocate storage variable for temp-quantities
+        if (options%time_options%RK3) allocate(temp(domain%ims:domain%ime,domain%kms:domain%kme,domain%jms:domain%jme))
 
-
-        if (options%physics%advection==kADV_UPWIND) then
-            call upwind(domain,options,dt)
+        if (options%physics%advection==kADV_STD) then
+            call adv_std_compute_wind(domain,options,dt)
         else if(options%physics%advection==kADV_MPDATA) then
-            call mpdata(domain,options,dt)
+            call mpdata_compute_wind(domain,options,dt)
         endif
+        
+
+        !Loop through all vars to advect
+                
+        ! make sure the dictionary is reset to point to the first variable
+        call domain%adv_vars%reset_iterator()
+        ! Now iterate through the dictionary as long as there are more elements present
+        do while (domain%adv_vars%has_more_elements())
+            ! get the next variable
+            var_to_advect = domain%adv_vars%next()
+
+            if (options%time_options%RK3) then
```

Comment:
let's talk, I'm not sure why you are making RK3 a time_option instead of an advection option. I realize it can be used in other places too, I'm just not clear how you are thinking of it here.

Reply:
The thought here comes from the WRF user guide, where there are other things calculated on the RK3 time step. So I figured it should be made universal, as it is not intrinsically tied to advection. In the current implementation, it is only used for advection, but based on the concept of what it is I thought it made sense to have it be a time option. It also affects the CFL criterion the user chooses, so from a UI perspective I think it is nice that the user changes the RK3 option right next to where the CFL options are changed, as a reminder to adapt them as necessary.